src/atem.ts

```diff
@@ -554,4 +605,104 @@ export class Atem extends EventEmitter {
 			delete this._sentQueue[trackingId]
 		}
 	}
+
+	/**
+	 * Helper method used by listVisibleInputs.
+	 * This got broken out into its own method because
+	 * it gets called multiple times, and gets called in a loop.
+	 * Breaking it out made listVisibleInputs much easier to read.
+	 */
+	private _calcActiveMeInputs (mode: 'program' | 'preview', meId: number): number[] {
+		const inputs = new Set<number>()
+		const meRef = this.state.video.getMe(meId)
+
+		if (mode === 'preview') {
+			if (meRef.transitionProperties.selection & 1) {
+				inputs.add(meRef.previewInput)
+			}
+		} else {
+			inputs.add(meRef.programInput)
+		}
+
+		// Upstream Keyers
+		Object.values(meRef.upstreamKeyers).filter(usk => {
+			// Pretty gross bitwise operations in this next line, be warned.
+			// TODO(Lange): This feels fragile but I'm unsure how to make it more robust.
+			const isPartOfTransition = meRef.transitionProperties.selection & (2 ** (usk.upstreamKeyerId + 1))
```

Comment:
I dont think this can be changed much, bitwise operations is just how something like this has to be done. 
I noramlly see this form though, so to me this is immediately more recognisable `& (1 << (usk.upstreamKeyerId + 1))`. Not sure if it is overall better though.

Other than that perhaps just pull that part into a variable to explain it a little better:
```
const keyerMask = 1 << (usk.upstreamKeyerId + 1)
const isPartOfTransition = meRef.transitionProperties.selection & keyerMask
```